### PR TITLE
Add Prettier, npm-run-all, Husky and update Travis tasks fix #54

### LINKED ISSUE #54: Add prettier formatter
https://gist.github.com/alexdiliberto/bdb0096aa63dabc1d6f389df564fc10c
https://github.com/ember-template-lint/ember-template-lint-plugin-prettier

### PR DESCRIPTION
**Pull request by zoltan-nz**
- Add Prettier to format code
- Handlebar support is limited/experimental, not reliable, so there isn't proper hbs formatting at the moment
- Use `npm-run-all`, `run-s` and `run-p` to run tasks.
- Add Husky to run linter, formatter check and test before commit. (Maybe running test is overkill we can remove it.)

- Coverall and codacy integration is turned off in Travis, because something wrong with the parser.

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
 
     {{content-for "head"}}
 
-    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/vendor.css">
-    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/library-app.css">
+    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/vendor.css" />
+    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/library-app.css" />
```
Comment (tkraak) [inline app/index.html]: I wonder why `prettier` is adding the closing `/` to `meta` and `link` here. None of the examples shown in the [Living HTML Standard](https://html.spec.whatwg.org/#the-meta-element) use the `/`.

https://html.spec.whatwg.org/#the-meta-element
https://html.spec.whatwg.org/#the-link-element

File: package.json [hunk 2]
```diff
@@ -69,5 +78,10 @@
   },
   "ember": {
     "edition": "octane"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "run-p lint:js:check format:check test"
```
Comment (tkraak) [inline package.json]: Should we do `pre-push` instead of `pre-commit` here?

File: .travis.yml [hunk 3]
```diff
@@ -22,10 +18,9 @@ branches:
...
-  - npm run test:coverage
+  - yarn ci
 
-after_script:
-  - npm run coveralls
-  - npm run codacy-coverage
+# @TODO Turn back after coveralls and codacy implementation are fixed
+#after_script:
+#  - yarn coveralls
+#  - yarn codacy-coverage
```
Comment (tkraak) [inline .travis.yml]: Do the `coveralls` and `codacy` implementations need to be fixed on our side? If so, let's add an issue once we merge this PR. Thanks.

File: package.json (hunk 2 above)
Comment (zoltan-nz) [reply package.json]: Good idea.

File: .travis.yml (hunk 3 above)
Comment (zoltan-nz) [reply .travis.yml]: Ticket is opened, so we can investigate. https://github.com/zoltan-nz/library-app/issues/66

File: app/index.html (hunk 1 above)
Comment (zoltan-nz) [reply app/index.html]: There is some discussion about it here https://github.com/prettier/prettier/issues/5246

The main point of using Prettier, that the most of the formatting decision is made by the Prettier community, so a dev team doesn't have to worry about formatting decision. ;) It looks to me that html is right with or without closing tag.

I prefer to customise prettier rules as minimal as possible. So if Prettier prefer this, I would suggest, let's keep in this way. :)

**Pull request merged**